service/controller/app/v1/resource/chartcrd/create.go

```diff
@@ -0,0 +1,61 @@
+package chartcrd
+
+import (
+	"context"
+	"time"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/app-operator/service/controller/app/v1/key"
+	"github.com/giantswarm/backoff"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/client/k8scrdclient"
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+
+	"github.com/giantswarm/app-operator/service/controller/app/v1/controllercontext"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	cr, err := key.ToCustomResource(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	secretName := key.SecretName(cr)
+
+	if secretName != "" {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "ensuring chart crd creation on tenant cluster")
+
+		ctlCtx, err := controllercontext.FromContext(ctx)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		if err != nil {
+			return microerror.Mask(err)
+		}
```

Comment:
I think that error handling here is not necessary.

Reply:
Thanks, I will delete this part.